test/security/testrunners/filepermissiontestrunner.go

```diff
@@ -0,0 +1,108 @@
+package testrunners
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/rule"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"log"
+	"strings"
+	"time"
+)
+
+type FilePermissionTestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+var _ test_runner.ITestRunner = (*FilePermissionTestRunner)(nil)
+
+const agentConfigPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const agentConfigOnlyRootRead = "-rw-r--r-- 1 root root "
+
+var (
+	onlyRootReadExactMatchRule = rule.Rule{
+		Conditions: []rule.ICondition{
+			&rule.ExactMatch{ExpectedValue: agentConfigOnlyRootRead},
+		},
+	}
+)
+
+var testCases = map[string]rule.Rule{
+	agentConfigPath: onlyRootReadExactMatchRule,
```

Comment:
What is this testing? `/opt/aws/amazon-cloudwatch-agent/bin/config.json` is not a reserved path for the agent outside of being the destination that the config-wizard writes the generated configuration to. Agent configuration files live in `/opt/aws/amazon-cloudwatch-agent/etc/`.

Reply:
I used the destination we copied config into in our tests. Was that wrong too?

Reply:
We're copying from the test into that location and then starting the agent with it. At translation time, the agent is going to take that JSON config and bring it to /etc/ for staging, then translate into TOML. If we're verifying file permissions, we really want to validate `/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml`

Reply:
There's a separate task for that one. (both toml and yaml). Are you saying this one is not worth validating?

Reply:
Yes I'm saying that the input config file doesn't matter. We can sync on that

Reply:
we talked about this a lot & made changes